### PR TITLE
Remove unnecessary forcing to univariate

### LINKED ISSUE #38012: Normal Form fails with single variable polynomials
Pull request at #38151 
### Steps To Reproduce

R.<z>=PolynomialRing(QQ)
A=AffineSpace(R)
f=DynamicalSystem_affine(z^2+z+1).homogenize(1)
f.normal_form()

### Expected Behavior

Should produce the normal form

### Actual Behavior

AttributeError: 'sage.rings.polynomial.polynomial_rational_flint.Polynomial_rational_flint' object has no attribute 'variable'

### Additional Information

_No response_

### Environment

```markdown
- **OS**: Ubuntu
- **Sage Version**: 10.0
-
```

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

### PR DESCRIPTION
**Pull request by Nathabolin**
Fixes #38012
by not forcing input to be uni-variate if already uni-variate

### :memo: Checklist

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

**Review by bhutz — CHANGES_REQUESTED** [review]
Add that this doctest is fixing issue 38012.

add white space around '+' in the example (line 8446)

File: src/sage/dynamics/arithmetic_dynamics/projective_ds.py [hunk 1]
```diff
@@ -8440,6 +8440,14 @@ def normal_form(self, return_conjugation=False):
               To:   Finite Field in z2 of size 3^2
               Defn: 1 |--> 1
 
+        Fixes issue 38012 by not forcing univariate polynomial to be univariate::
```
Comment (S17A05) [inline src/sage/dynamics/arithmetic_dynamics/projective_ds.py]:
```suggestion
        Fixes :issue:`38012` by not forcing univariate polynomial to be univariate::
```

File: src/sage/dynamics/arithmetic_dynamics/projective_ds.py [hunk 2]
```diff
@@ -8468,7 +8476,8 @@ def normal_form(self, return_conjugation=False):
         #we find one and not go all the way to the splitting field
         i = 0
         if G.degree() != 0:
-            G = G.polynomial(G.variable(0))
+            if is_MPolynomialRing(G.parent()):
+              G = G.polynomial(G.variable(0))
```
Comment (S17A05) [inline src/sage/dynamics/arithmetic_dynamics/projective_ds.py]:
```suggestion
            if is_MPolynomialRing(G.parent()):
                G = G.polynomial(G.variable(0))
```

**Comment by mkoeppe** [comment]
Hints:
- Please change the title of the PR so we don't have to look up the Issue by number.
- https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

**Comment by S17A05** [comment]
What @mkoeppe was referring to with the second hint: You should write "Fixes #38012" (without the "issue") in the PR text body, so that the issue gets linked to this PR.

**Review by bhutz — APPROVED** [review]
This good to me and all tests are passing and the docs build.

**Comment by Nathabolin** [comment]
What needs to be done for this to be accepted? or is it accepted?

**Comment by S17A05** [comment]
> What needs to be done for this to be accepted? or is it accepted?

This is accepted, it just needs some time to be merged since the release of Sage 10.4 is currently being prepared, and afaik only urgent stuff is getting merged right now. Also merging the new develop-branch into your PR resets the "positive review"-label, so you should avoid it unless there are merge conflicts that need to be resolved.

**Comment by Nathabolin** [comment]
Good to know thank you so much

File: src/sage/dynamics/arithmetic_dynamics/projective_ds.py [hunk 3]
```diff
@@ -8437,6 +8437,14 @@ def normal_form(self, return_conjugation=False):
               To:   Finite Field in z2 of size 3^2
               Defn: 1 |--> 1
 
+        Fixes :issue:`38012` by not forcing univariate polynomial to be univariate::
+
+            sage: R.<z> = PolynomialRing(QQ)
```
Comment (kwankyu) [inline src/sage/dynamics/arithmetic_dynamics/projective_ds.py:8442]: The comment says an implementation detail that is not understandable by a user. The usual way to introduce a test (example) for a fix is
```
TEST:

Check that :issue:`38012` is fixed::

    sage: R.<z> = PolynomialRing(QQ)

```

**Comment by kwankyu** [comment]
Please correct the PR title.

**Pull request merged**